code/src/test_usart.cpp

```diff
@@ -0,0 +1,57 @@
+#include <test_usart.hpp>
+
+namespace r2d2 {
+
+    test_usart_c::test_usart_c(){}
+
+    void test_usart_c::enable(){
+        //Not needed in test implementation
+    }
+
+    void test_usart_c::disable(){
+        //Not needed in test implementation
+    }
+
+    bool test_usart_c::send(const uint8_t c) {
+        return true;
+        //Not needed in test implementation
+    }
+
+    void test_usart_c::putc(char c) {
+        send(c);
+    }
+
+    uint8_t test_usart_c::receive() {
+        return receive_buffer.copy_and_pop();
+    }
+
+    bool test_usart_c::char_available() {
```

Comment:
What is the difference between ```char_available()``` and ```available()```?

Reply:
char_available returns true if char is available, available returns the amound of bytes available

Reply:
But both ```char_available()``` and ```available()``` return ```!receive_buffer.empty()```